newIDE/app/src/VariablesList/VariableRow.js

```diff
@@ -109,38 +145,68 @@ const VariableRow = ({
         />
       </TreeTableCell>
     );
-  } else {
-    columns.push(
-      <TreeTableCell
-        expand
-        key="value"
-        style={limitEditing ? styles.fadedButton : undefined}
-      >
-        <Text noMargin>(Structure)</Text>
-      </TreeTableCell>
-    );
   }
   columns.push(
     <TreeTableCell key="tools" style={styles.toolColumn}>
-      {origin === 'inherited' && depth === 0 && (
+      {origin === 'inherited' && depth === 0 ? (
         <IconButton
           size="small"
           onClick={onResetToDefaultValue}
-          style={isStructure ? undefined : styles.fadedButton}
+          style={isStructural ? undefined : styles.fadedButton}
           tooltip={t`Reset`}
         >
           <Replay />
         </IconButton>
-      )}
-      {!(origin === 'inherited' && depth === 0) && origin !== 'parent' && (
-        <IconButton
-          size="small"
-          onClick={onAddChild}
-          style={isStructure ? undefined : styles.fadedButton}
-          tooltip={t`Add child variable`}
-        >
-          <AddCircle />
-        </IconButton>
+      ) : (
+        origin !== 'parent' && (
+          <>
```

Comment:
You can use `ElementWithMenu` to get a menu that will be using material-ui on the web and a native menu on Electron:

https://github.com/4ian/GDevelop/blob/664b18556e9339322f39a071bb3e93642643b6d7/newIDE/app/src/LayersList/LayerRow.js#L70-L100

That should allow you to remove all the complexity related to the popover

Reply:
I think the popover looks better and has also the advantage of being compact  

![image](https://user-images.githubusercontent.com/19349038/102469288-dc276f80-4052-11eb-8d45-6f0cba3d1347.png)  

![image](https://user-images.githubusercontent.com/19349038/102470373-2ceb9800-4054-11eb-9575-abcf631f0933.png)

Reply:
It might indeed look a bit better to a user familiar with these notions, but in this case I would rather improve the UI building blocks in a separate PR to allow to have context menus composed of icons only, or on Electron to allow setting icons in addition to the labels.

It would also go beyond this very specific menu: if your menu has a need for displaying only icons, then presumably other menus would benefit from it? And so this should be part of a larger change to bring icons only menu to the app.

But I'm a bit worried of accessibility: you have to hover each icon to learn what they do. We're also not a in space constrained situation, so I don't see any reason not to display the labels - they are providing useful information and allow to understand your choice in 2 seconds (the time to read), rather than 5 seconds (the time to try to understand the icon, and then to hover them to have confirmation).